src/app_service/service/keycard/service.nim

```diff
@@ -97,9 +97,13 @@ QtObject:
   proc init*(self: Service) =
     if self.doLogging:
       debug "init keycard using ", pairingsJson=status_const.KEYCARDPAIRINGDATAFILE
-    let initResp = keycard_go.keycardInitFlow(status_const.KEYCARDPAIRINGDATAFILE)
-    if self.doLogging:
-      debug "initialization response: ", initResp
+    # This code must not be executed on Darwin amd64
+    # since go 1.21 Intel macs are not able to handle nil cStrings from go
+    # resulting in bad flushGen 12 in prepareForSweep; sweepgen 0
+    if not (status_const.IS_MACOS and status_const.IS_INTEL):
+      let initResp = keycard_go.keycardInitFlow(status_const.KEYCARDPAIRINGDATAFILE)
```

Comment:
We have to find a way and do this call even for Mac Intel.

I haven't had any issues on Mac M2, also when I was using Mac Intel, didn't have any issues, but the Go version there at that time was 1.20.4.

When we're in this code, it should already be a `string` response, that's why I'm wondering isn't something up to the code here https://github.com/status-im/nim-keycard-go/blob/develop/keycard_go.nim#L5-L8 where map Go response and working with raw pointers.

Reply:
Hmm.
You're probably right.

At this point I'd be happy if someone from Desktop team takes over the work done in this PR.
All I know is that skipping `keycardInitFlow` and `keycardCancelFlow` causes the crash to go away.
 
Reverting go upgrade is a big hassle because then we will have to downgrade go version of `status-go`, `go-waku` and `status-mobile` so in my opinion we should spend time in finding a fix for this issue.

Reply:
Will try to setup Mac Intel later and reproduce the issue, if I can reproduce it, will try to fix it.

@siddarthkay thanks a lot for help on this, you localized a problem very well.